GitUI/CommandsDialogs/RevisionFileTree.cs

```diff
@@ -493,14 +494,13 @@ private void FileTreeContextMenu_Opening(object sender, System.ComponentModel.Ca
             }
 
             saveAsToolStripMenuItem.Visible = isFile;
-            resetToThisRevisionToolStripMenuItem.Visible = isFile && !Module.IsBareRepository();
-            toolStripSeparatorFileSystemActions.Visible = isFile;
+            resetToThisRevisionToolStripMenuItem.Visible = isFileOrFolder && !Module.IsBareRepository();
+            toolStripSeparatorFileSystemActions.Visible = isFileOrFolder;
 
-            fileHistoryToolStripMenuItem.Enabled = itemSelected;
             copyFilenameToClipboardToolStripMenuItem.Visible = itemSelected;
             fileTreeOpenContainingFolderToolStripMenuItem.Enabled = isExistingFileOrDirectory;
             fileTreeArchiveToolStripMenuItem.Enabled = itemSelected;
-            fileTreeCleanWorkingTreeToolStripMenuItem.Visible = isFolder;
+            fileTreeCleanWorkingTreeToolStripMenuItem.Visible = isFileOrFolder;
```

Comment:
either if I miss something but you could clean a working tree, lony if the item is a folder, no!?! So I think it is not good. You should keep `isFolder`.

Reply:
If you clean on a file, you clean the directory, so it works. Clean-on-files was working like that in 2.50 but I agree the functionality is weird.